### PR TITLE
Add support for namespaces (and permissions)

### PR DESCRIPTION
**Pull request by abought**
Refs https://openscience.atlassian.net/browse/LEI-37
## Purpose

We would like to manage permissions for namespaces and collections.

screen shot 2016-02-02 at 1 04 59 pm
## Summary of changes
- Move behavior from application adapter/serializer to collection mixins- as we add more types, if doesn't make sense to treat everything like a collection
- Add models for namespaces
- Add scaffold for project settings page, with ability to edit permissions for entire collection
- Improve serializer support for write operations to server (field exclusion etc)
## Sample tests
- Navigate to the project settings page. Add and remove a user (for now, manually enter the entire permissions string)
  - Then try to access any collection using the new user (eg OSF or Jam ID of a different account)
- Change the project-settings page to fetch a single collection (`this.store.find('collection', 'experiments')`). Use that page to edit permissions on the collection.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -11,24 +11,24 @@ charset = utf-8
 trim_trailing_whitespace = true
```
Comment (samchrisinger) [inline .editorconfig:11]: Can you please remove this and add it to either your personal gitignore or the one for this repo?

Comment (samchrisinger) [inline .editorconfig:11]: ^ This is an ember-cli-build added file and should be kept around. h/t Andy

**Comment by samchrisinger** [comment]
Sorry, wrong button

**Comment by abought** [comment]
As a side note, it appears that this refactor left out the adapter for accounts. If you need it before the next PR, it can be generated as follows:

`ember g adapter account`

(Content of file: straight copy-paste of another collection adapter, like `adapters/admin.js`)

**Pull request merged**